libmultilabel/model.py

```diff
@@ -1,218 +1,108 @@
+    """Abstract class handling Pytorch Lightning training flow"""
+    def __init__(self, config, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if isinstance(config, Namespace):
+            config = vars(config)
+        if isinstance(config, dict):
+            config = AttributeDict(config)
         self.config = config
-        self.device = config.device
-        self.start_epoch = 0
-
-        if ckpt:
-            self.config.run_name = ckpt['run_name']
-            self.word_dict = ckpt['word_dict']
-            self.classes = ckpt['classes']
-            self.best_metric = ckpt['best_metric']
-            self.start_epoch = ckpt['epoch']
-        else:
-            self.word_dict = word_dict
-            self.classes = classes
-            self.start_epoch = 0
-            self.best_metric = 0
-
-        self.config.num_classes = len(self.classes)
-
-        embed_vecs = self.word_dict.vectors
-        self.network = getattr(networks, config.model_name)(config, embed_vecs).to(self.device)
-        self.init_optimizer()
-
-        if ckpt:
-            self.network.load_state_dict(ckpt['state_dict'])
-            self.optimizer.load_state_dict(ckpt['optimizer'])
-        elif config.init_weight is not None:
-            init_weight = networks.get_init_weight_func(config)
-            self.network.apply(init_weight)
 
-    def init_optimizer(self, optimizer=None):
+    def configure_optimizers(self):
         """Initialize an optimizer for the free parameters of the network.
-        Args:
-            state_dict: network parameters
         """
-        parameters = [p for p in self.network.parameters() if p.requires_grad]
-        optimizer_name = optimizer or self.config.optimizer
+        parameters = [p for p in self.parameters() if p.requires_grad]
+        optimizer_name = self.config.optimizer
         if optimizer_name == 'sgd':
-            self.optimizer = optim.SGD(parameters, self.config.learning_rate,
-                                       momentum=self.config.momentum,
-                                       weight_decay=self.config.weight_decay)
+            optimizer = optim.SGD(parameters, self.config.learning_rate,
+                                  momentum=self.config.momentum,
+                                  weight_decay=self.config.weight_decay)
         elif optimizer_name == 'adam':
-            self.optimizer = optim.Adam(parameters, weight_decay=self.config.weight_decay, lr=self.config.learning_rate)
-
+            optimizer = optim.Adam(
+                parameters, weight_decay=self.config.weight_decay, lr=self.config.learning_rate)
+        elif optimizer_name == 'adamw':
+            optimizer = optim.AdamW(
+                parameters, weight_decay=self.config.weight_decay, lr=self.config.learning_rate)
         else:
-            raise RuntimeError('Unsupported optimizer: %s' % self.config.optimizer)
+            raise RuntimeError('Unsupported optimizer: %s' %
```

Comment:
why this change?

Reply:
Both change are caused by editor auto-formatting. Fixed.